app/controllers/jobs_controller.rb

```diff
@@ -21,24 +24,58 @@ def create
   end
 
   def show
-    @job = Job.find(params[:id])
+    @comment = Comment.new
+    @comment.job_id = @job.id
   end
 
   def edit
-    # implement on your own!
   end
 
   def update
-    # implement on your own!
+    @job.update(job_params)
+    if @job.save
+      flash[:success] = "#{@job.title} updated!"
+      redirect_to company_job_path(@company, @job)
+    else
+      render :edit
+    end
   end
 
   def destroy
-    # implement on your own!
+    @job.destroy
+
+    flash[:success] = "#{@job.title} at #{@job.company.name} was successfully deleted!"
+    redirect_to company_jobs_path(@company)
+  end
+
+  def jobs
+    if params[:sort] == "location"
```

Comment:
your views are going to go off the method/action name so if i didnt want to explicitly render index2, i could name this method index2 and it would look for a view of that name.